### PR TITLE
TestCheckPerms: utilize pipe to read stderr

### PR DESCRIPTION
**Pull request by zhijianli88**
Refer to the stderr manpage:
$ man 3 stderr
*Note that mixing use of FILEs and raw file descriptors can produce unexpected results and should generally be avoided.*

And actually, we noticed that the warning() will output the message to
stdout instead of stderr sometimes.

lizj@FNSTPC:~/workspace/k8s/helm$ while true; do timeout 1m go test -count=1  -run TestCheckPerms ./cmd/helm -v 2>/dev/null; done
=== RUN   TestCheckPerms
--- PASS: TestCheckPerms (0.00s)
PASS
ok  	helm.sh/helm/v3/cmd/helm	0.028s
=== RUN   TestCheckPerms
--- PASS: TestCheckPerms (0.00s)
PASS
ok  	helm.sh/helm/v3/cmd/helm	0.027s
=== RUN   TestCheckPerms
--- PASS: TestCheckPerms (0.00s)
PASS
ok  	helm.sh/helm/v3/cmd/helm	0.028s
=== RUN   TestCheckPerms
--- PASS: TestCheckPerms (0.00s)
PASS
ok  	helm.sh/helm/v3/cmd/helm	0.029s
=== RUN   TestCheckPerms
--- PASS: TestCheckPerms (0.00s)
PASS
ok  	helm.sh/helm/v3/cmd/helm	0.029s
=== RUN   TestCheckPerms
--- PASS: TestCheckPerms (0.00s)
PASS
ok  	helm.sh/helm/v3/cmd/helm	0.028s
=== RUN   TestCheckPerms
--- PASS: TestCheckPerms (0.00s)
PASS
ok  	helm.sh/helm/v3/cmd/helm	0.030s
=== RUN   TestCheckPerms
WARNING: Kubernetes configuration file is group-readable. This is insecure. Location: /tmp/helmtest093620773/testconfig
=== RUN   TestCheckPerms
WARNING: Kubernetes configuration file is group-readable. This is insecure. Location: /tmp/helmtest083469215/testconfig
=== RUN   TestCheckPerms
WARNING: Kubernetes configuration file is group-readable. This is insecure. Location: /tmp/helmtest101343249/testconfig
=== RUN   TestCheckPerms
--- PASS: TestCheckPerms (0.00s)
PASS
ok  	helm.sh/helm/v3/cmd/helm	0.032s
=== RUN   TestCheckPerms
--- PASS: TestCheckPerms (0.00s)
PASS
ok  	helm.sh/helm/v3/cmd/helm	0.040s
=== RUN   TestCheckPerms
--- PASS: TestCheckPerms (0.00s)
PASS
ok  	helm.sh/helm/v3/cmd/helm	0.031s
=== RUN   TestCheckPerms
WARNING: Kubernetes configuration file is group-readable. This is insecure. Location: /tmp/helmtest706352639/testconfig

Signed-off-by: Li Zhijian <lizhijian@cn.fujitsu.com>
Signed-off-by: Lu Fengqi <lufq.fnst@cn.fujitsu.com>

**What this PR does / why we need it**:

**Special notes for your reviewer**:

**If applicable**:
- [ ] this PR contains documentation
- [ ] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

### CODE REVIEW

File: cmd/helm/root_unix_test.go [hunk 1]
```diff
@@ -19,23 +19,36 @@ limitations under the License.
...
-	defer stderr.Close()
-	reader := bufio.NewReader(stderr)
 
+	stderr := os.Stderr
+	os.Stderr = w
+	defer func() {
+		os.Stderr = stderr
+	}()
+
+	f()
```
Comment (littleroad) [inline cmd/helm/root_unix_test.go:43]: Why not directly call checkPerm here?

Comment (zhijianli88) [reply cmd/helm/root_unix_test.go]: make sense !

**Review by bacongobbler — APPROVED** [review]
Thanks.

**Comment by bacongobbler** [comment]
cc @adamreese

**Review by mattfarina — APPROVED** [review]
lgtm. This also fixes an error on some macos systems when running the test suite.

**Pull request merged**